server/server.go

```diff
@@ -452,6 +446,56 @@ func (h RequestHandler) NewThreadRoute(res http.ResponseWriter, req *http.Reques
 	}
 }
 
+func (h RequestHandler) DeletePostRoute(res http.ResponseWriter, req *http.Request) {
+	if req.Method == "GET" {
+		IndexRedirect(res, req)
+		return
+	}
+	threadURL := req.PostFormValue("thread")
+	postid, ok := util.GetURLPortion(req, 3)
+	loggedIn, userid := h.IsLoggedIn(req)
+
+	if !loggedIn || !ok {
+		title := "Unaccepted request"
+		data := GenericMessageData{
+			Title:       title,
+			Message:     "The post you tried to delete was not found, or you were not allowed to delete it",
+			LinkMessage: "Go back to",
+			Link:        threadURL,
+			LinkText:    "the thread",
+		}
+		h.renderView(res, "generic-message", TemplateData{Data: data, LoggedIn: loggedIn})
+		return
+	}
+
+	post, err := h.db.GetPost(postid)
+	authorized := post.AuthorID == userid
+	if !authorized || err != nil {
+		title := "Unaccepted request"
+		data := GenericMessageData{
+			Title:       title,
+			Message:     "You were not allowed to delete the post",
+			LinkMessage: "Go back to",
+			Link:        threadURL,
+			LinkText:    "the thread",
+		}
+		h.renderView(res, "generic-message", TemplateData{Data: data, LoggedIn: loggedIn})
+		return
+	}
+
+	switch req.Method {
+	case "POST":
+		if authorized {
+			// TODO (2022-01-12): receive error and render it
+      err = h.db.DeletePost(postid)
```

Comment:
wonky spacing, gofmt? I can help add ci linting and automation if you want :)

Reply:
nah no ci linting, thanks. i just gotta get my shit together lolol

(i've got a nice pre-commit hook hiding on my work computer i think :)